modules/sequenced_capture_executor.py

```diff
@@ -666,16 +666,20 @@ def _capture(
 
                 start_ts = time.time()
                 stop_ts = start_ts + duration_sec
+                expected_frames = fps * duration_sec
+                captured_frames = 0
                 seconds_per_frame = 1.0 / fps
 
-                while time.time() < stop_ts:
+                while time.time() < stop_ts or captured_frames < expected_frames:
                     if 'update_scope_display' in self._callbacks:
                         self._callbacks['update_scope_display']()
                         
                     # Currently only support 8-bit images for video
                     force_to_8bit = True
                     image = self._scope.get_image(force_to_8bit=force_to_8bit)
 
+                    captured_frames += 1
```

Comment:
Is it correct to increment captured_frames here? Or should it be underneath video_writer.add_frame(image=image)? Wasn't fully clear on what the if statement type(image) == np.ndarray was looking for.

Reply:
The check for `type(image) == np.ndarray` is to protect against the case where the camera is unable to grab an image (and returns False).

I think it's okay to add `captured_frames` where you did.  There might be corner cases where the duration ends up longer than intended because failed frames don't count towards the time.  It may be better to calculate the `start_ts` and `stop_ts` after the first frame is grabbed rather than ahead of time, instead of using the captured_frames counter.  Dropped frames will cause it to run the capture for a longer 'calendar' time, even though the video time will be correct, if that makes sense.

I think this is good for now though as a start.